src/mlpack/methods/decision_tree/decision_tree.hpp

```diff
@@ -99,7 +100,7 @@ class DecisionTree :
    * @param maximumDepth Maximum depth for the tree.
    * @param dimensionSelector Instantiated dimension selection policy.
    */
-  template<typename MatType, typename LabelsType>
```

Comment:
why the `MatType` has been removed?

Reply:
I made the tempname Mattype global to make the training dataset a global variable for the whole class.
This will help in checking whether testing dataset will have same number of features as training.
Thus removed from here and added in template of whole class.

Reply:
I haven't reviewed the rest of the code yet, but let's not make that change.  There is no need to hold the entire training dataset in each node of the tree, or restrict the type of the matrix used for testing to be the same as the training matrix type.  (In fact holding the training dataset in each node of the tree is a huge no-no: you duplicate the dataset once for *every* node in the tree.  Further, please don't add any members to `DecisionTree`.  It's tuned to have minimal size.)

Reply:
Ok I will do some changes to solve this problem.

Reply:
hey @rcurtin Now is it ok ?
I have added a post function which will be called when the main function is recurcively called when going to other node.
This is to avoid duplicating the whole dataset in evey node.

Reply:
I still see that you've removed template parameters.  Let's not do that.